src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,114 @@
+import os
+import pathlib
+import tempfile
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    _dir: tempfile.TemporaryDirectory
+
+    def __init__(self):
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.mocks: dict[str, unittest.mock.patch] = {}
+        for label, path in paths.items():
+            mocked_path = f"{self.chroot}{path}"
+            pathlib.Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
+            self.mocks[label] = unittest.mock.patch(label, mocked_path)
```

Comment:
When called “target” instead of “label”, it conveys the idea clearer.

```suggestion
            self.mocks[target] = unittest.mock.patch(target, mocked_path)
```